test/uk/gov/hmrc/gform/eval/ExpressionResultSpec.scala

```diff
@@ -267,4 +267,32 @@ class ExpressionResultSpec extends FunSuite {
       assertEquals(lhs after rhs, expected)
     }
   }
+
+  test("ListResult and NumberResult operations") {
+    val nr1 = NumberResult(1)
+    val nr2 = NumberResult(2)
+    val lr12 = ListResult(List(nr1, nr2))
+
+    assertEquals(nr2 + lr12, ListResult(List(NumberResult(3), NumberResult(4))))
+    assertEquals(lr12 + nr2, ListResult(List(NumberResult(3), NumberResult(4))))
+    assertEquals(nr2 - lr12, ListResult(List(NumberResult(1), NumberResult(0))))
+    assertEquals(lr12 - nr2, ListResult(List(NumberResult(-1), NumberResult(0))))
+    assertEquals(nr2 * lr12, ListResult(List(NumberResult(2), NumberResult(4))))
+    assertEquals(lr12 * nr2, ListResult(List(NumberResult(2), NumberResult(4))))
+    assertEquals(nr2 / lr12, ListResult(List(NumberResult(2), NumberResult(1))))
+    assertEquals(lr12 / nr2, ListResult(List(NumberResult(0.5), NumberResult(1))))
+  }
+
+  test("ListResult and ListResult operations") {
+    val nr1 = NumberResult(1)
+    val nr2 = NumberResult(2)
+    val nr3 = NumberResult(3)
+    val lr12 = ListResult(List(nr1, nr2))
+    val lr13 = ListResult(List(nr1, nr3))
+
+    assertEquals(lr13 + lr12, ListResult(List(NumberResult(2), NumberResult(5))))
+    assertEquals(lr13 - lr12, ListResult(List(NumberResult(0), NumberResult(1))))
+    assertEquals(lr13 * lr12, ListResult(List(NumberResult(1), NumberResult(6))))
+    assertEquals(lr13 / lr12, ListResult(List(NumberResult(1), NumberResult(1.5))))
+  }
```

Comment:
It would be great to cover interaction with `Hidden` and perhaps other cases too.

Reply:
I tried to emulate ListRestult behavior on Hidden to how it works with NumberResults
```
Hidden + nr1
// res42: uk.gov.hmrc.gform.eval.ExpressionResult = NumberResult(1)
Hidden - nr1
// res43: uk.gov.hmrc.gform.eval.ExpressionResult = NumberResult(1)
Hidden * nr1
// res44: uk.gov.hmrc.gform.eval.ExpressionResult = NumberResult(0)
Hidden / nr1
// res45: uk.gov.hmrc.gform.eval.ExpressionResult = NumberResult(0)


nr1 + Hidden
// res46: uk.gov.hmrc.gform.eval.ExpressionResult = NumberResult(1)
nr1 - Hidden
// res47: uk.gov.hmrc.gform.eval.ExpressionResult = NumberResult(1)
nr1 * Hidden
// res48: uk.gov.hmrc.gform.eval.ExpressionResult = NumberResult(0)
nr1 / Hidden
// res49: uk.gov.hmrc.gform.eval.ExpressionResult = NumberResult(0)
```

and this is how it works with lr12 = ListResult(List(NumberResult(1), NumberResult(2))))
```
Hidden + lr12
// res50: uk.gov.hmrc.gform.eval.ExpressionResult = ListResult(List(NumberResult(1), NumberResult(2)))
Hidden - lr12
// res51: uk.gov.hmrc.gform.eval.ExpressionResult = ListResult(List(NumberResult(1), NumberResult(2)))
Hidden * lr12
// res52: uk.gov.hmrc.gform.eval.ExpressionResult = ListResult(List(NumberResult(0), NumberResult(0)))
Hidden / lr12
// res53: uk.gov.hmrc.gform.eval.ExpressionResult = ListResult(List(NumberResult(0), NumberResult(0)))


lr12 + Hidden
// res54: uk.gov.hmrc.gform.eval.ExpressionResult = ListResult(List(NumberResult(1), NumberResult(2)))
lr12 - Hidden
// res55: uk.gov.hmrc.gform.eval.ExpressionResult = ListResult(List(NumberResult(1), NumberResult(2)))
lr12 * Hidden
// res56: uk.gov.hmrc.gform.eval.ExpressionResult = ListResult(List(NumberResult(0), NumberResult(0)))
lr12 / Hidden
// res57: uk.gov.hmrc.gform.eval.ExpressionResult = ListResult(List(NumberResult(0), NumberResult(0)))
```